src/arrayMethodSort.js

```diff
@@ -5,7 +5,23 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const array = this;
+
+    for (let i = 0; i < array.length - 1; i++) {
+      for (let j = 0; j < array.length - i - 1; j++) {
+        const a = array[j];
+        const b = array[j + 1];
+        const compareValue = compareFunction
+          ? compareFunction(a, b)
+          : String(a) > String(b);
```

Comment:
it's better to `String(a) > String(b)` as default value to the compareFunction argument